src/components/GoalActivityFeed.tsx

```diff
@@ -0,0 +1,191 @@
+            onGoalDependencyAdd,
+            onGoalDependencyRemove,
+            lastStateComment,
+            highlightCommentId,
+        } = useGoalResource(
+            {
+                id: goal?.id,
+                stateId: goal?.stateId,
+                reactions: goal?.reactions,
+                comments: goal?.comments,
+            },
+            {
+                invalidate: {
+                    getById: shortId,
+                },
+            },
+        );
+
+        const commentDraft = resolveGoalCommentDraft(goal?.id);
+
+        const onGoalDeleteConfirm = useCallback(() => {
+            onGoalDelete();
+            onDelete?.();
+        }, [onGoalDelete, onDelete]);
+
+        const onGoalCommentSubmit = useCallback(
+            (value: NonNullable<GoalByIdReturnType>['comments'][number]) => {
+                return value.activity?.id === activityId ? onGoalCommentUpdate(value.id) : undefined;
+            },
+            [onGoalCommentUpdate, activityId],
+        );
+
+        return (
+            <>
+                {nullable(goal, ({ _activityFeed, id, goalAchiveCriteria, _relations, project, _isEditable }) => (
+                    <GoalActivity
+                        ref={ref}
+                        feed={_activityFeed}
+                        header={
+                            <>
+                                {nullable(goalAchiveCriteria.length || _isEditable, () => (
+                                    <GoalCriteria
+                                        goalId={id}
+                                        criteriaList={goalAchiveCriteria}
+                                        onAddCriteria={onGoalCriteriaAdd}
+                                        onToggleCriteria={onGoalCriteriaToggle}
+                                        onRemoveCriteria={onGoalCriteriaRemove}
+                                        onConvertToGoal={onGoalCriteriaConvert}
+                                        onUpdateCriteria={onGoalCriteriaUpdate}
+                                        onClick={onGoalCriteriaClick}
+                                        canEdit={_isEditable}
+                                        renderTrigger={(props) =>
+                                            nullable(_isEditable, () => (
+                                                <AddCriteriaForm
+                                                    goalId={props.goalId}
+                                                    onSubmit={props.onSubmit}
+                                                    validityData={props.validityData}
+                                                />
+                                            ))
+                                        }
+                                    />
+                                ))}
+
+                                {_relations.map((deps) =>
+                                    nullable(deps.goals.length || _isEditable, () => (
```

Comment:
I'm not sure this logic is ok.

<img width="92" alt="image" src="https://github.com/taskany-inc/issues/assets/982072/c224c563-7074-41dc-9902-7ec9330add02">

Reply:
this logic is equivalent to the previous one, but its easier to understand